src/nft/strings.cc

```diff
@@ -487,6 +488,21 @@ Nft nft::strings::replace_reluctant_single_symbol(Symbol from_symbol, Symbol rep
 
 Nft ReluctantReplace::replace_regex(nfa::Nfa regex, const Word& replacement, Alphabet* alphabet,
                                               ReplaceMode replace_mode, Symbol begin_marker) {
+    // SMT-LIB theory Strings (Unicode Strings): Semantics for matching empty words in replace regex functions:
+    // ; (str.replace_re s r t) is the string obtained by replacing the
+    // ; shortest leftmost match of r in s, if any, by t.
+    // ; Note that if the language of r contains the empty string, 
+    // ; the result is to prepend t to s.
+    // (str.replace_re String RegLan String String) 
+    //
+    // ; (str.replace_re_all s r t) is the string obtained by replacing,
+    // ; left-to right, each shortest *non-empty* match of r in s by t.
+    // (str.replace_re_all String RegLan String String)
+    if (replace_mode == ReplaceMode::All) {
+        // Removing the empty string from the regex.
+        regex.unify_initial(true);
```

Comment:
You still need to remove the new initial state from the set of final states here.

Reply:
It was WIP. Fixed in the latest commit with new tests covering these edge cases.

Reply:
Ok, let me know when it is finished, so I can look at it again.